camel-core/src/main/java/org/apache/camel/model/ModelHelper.java

```diff
@@ -110,13 +163,154 @@ public static String dumpModelAsXml(CamelContext context, NamedNode definition)
+        }
+
+        // Restore namespaces to anything that's NamespaceAware
+        if (result instanceof RoutesDefinition) {
+            List<RouteDefinition> routes = ((RoutesDefinition) result).getRoutes();
+            for (RouteDefinition route : routes) {
+                applyNamespaces(route, namespaces);
+            }
+        } else if (result instanceof RouteDefinition) {
+            RouteDefinition route = (RouteDefinition) result;
+            applyNamespaces(route, namespaces);
+        }
+
         return type.cast(result);
     }
 
+    /**
+     * Marshal the xml to the model definition
+     *
+     * @param context the CamelContext, if <tt>null</tt> then {@link org.apache.camel.spi.ModelJAXBContextFactory} is not in use
+     * @param inputStream the xml stream
+     * @throws Exception is thrown if an error is encountered unmarshalling from xml to model
+     */
+    public static RoutesDefinition loadRoutesDefinition(CamelContext context, InputStream inputStream) throws Exception {
+        JAXBContext jaxbContext;
+        if (context == null) {
+            jaxbContext = createJAXBContext();
+        } else {
+            jaxbContext = context.getModelJAXBContextFactory().newJAXBContext();
+        }
+
+        Map<String, String> namespaces = new LinkedHashMap<>();
+        Document dom = context.getTypeConverter().mandatoryConvertTo(Document.class, inputStream);
+        extractNamespaces(dom, namespaces);
+
+        Binder<Node> binder = jaxbContext.createBinder();
+        Object result = binder.unmarshal(dom);
+
+        if (result == null) {
+            throw new JAXBException("Cannot unmarshal to RoutesDefinition using JAXB");
+        }
+
+        // can either be routes or a single route
+        RoutesDefinition answer;
+        if (result instanceof RouteDefinition) {
+            RouteDefinition route = (RouteDefinition) result;
+            answer = new RoutesDefinition();
+            applyNamespaces(route, namespaces);
+            answer.getRoutes().add(route);
+        } else if (result instanceof RoutesDefinition) {
+            answer = (RoutesDefinition) result;
+            for (RouteDefinition route : answer.getRoutes()) {
+                applyNamespaces(route, namespaces);
+            }
+        } else {
+            throw new IllegalArgumentException("Unmarshalled object is an unsupported type: " + ObjectHelper.className(result) + " -> " + result);
+        }
+
+        return answer;
+    }
+
+    private static void applyNamespaces(RouteDefinition route, Map<String, String> namespaces) {
+        Iterator<ExpressionNode> it = filterTypeInOutputs(route.getOutputs(), ExpressionNode.class);
+        while (it.hasNext()) {
+            ExpressionNode en = it.next();
+            ExpressionDefinition ed = en.getExpression();
+
+            NamespaceAware na = null;
+            Expression exp = ed.getExpressionValue();
+            if (exp != null && exp instanceof NamespaceAware) {
```

Comment:
you can check only exp instanceof NamespaceAware because "null instanceof NamespaceAware" is returning false